src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -369,6 +386,36 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [isMobile]);
 
+    useEffect(() => {
+      if (isMobile) {
+        return;
+      }
+
+      const totalActiveDrawersMinSize = getActiveDrawersTotalMinSize();
+      const activeNavigationWidth = navigationOpen ? navigationWidth : 0;
+      // collapsed content width is $space-layout-content-horizontal * 2 = 48px
+      const minContentVisibleWidth = 48;
+
+      const scrollWidth = activeNavigationWidth + minContentVisibleWidth + totalActiveDrawersMinSize;
+      const hasHorizontalScroll = scrollWidth > placement.inlineSize;
+      if (hasHorizontalScroll) {
+        if (navigationOpen) {
+          onNavigationToggle(false);
+          return;
+        }
+
+        closeFirstDrawer();
```

Comment:
Didn't we want to have an iteration here, "close until hasHorizontalScroll is false"?

Reply:
No, we don't, because closing a drawer triggers a change in the dependency array, which leads to recalculations until the horizontal scrollbar is no longer present

Reply:
Not instantly obvious, but okay